lib/SimpleHttpClient/SslClientConnection.cpp

```diff
@@ -446,48 +473,57 @@ bool SslClientConnection::writeClientConnection(void const* buffer,
     return false;
   }
 
-  int written = SSL_write(_ssl, buffer, (int)length);
-  int err = SSL_get_error(_ssl, written);
-  switch (err) {
-    case SSL_ERROR_NONE:
-      *bytesWritten = written;
+  int written = -1;
+  auto start = steady_clock::now();
```

Comment:
Is any modification to `writeClientConnection()` needed, given that the socket flags are only modified upon connection?
It would be better to minimize the changes in this PR so that only the connection part is changed.